### PR TITLE
[Improve]Ingress uses different versions of API according to differen…

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
## What changes were proposed in this pull request
close # #2698

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.apache.streampark.common.util.Logger
+
+import io.fabric8.kubernetes.client.DefaultKubernetesClient
+import org.apache.flink.client.program.ClusterClient
+
+import scala.language.postfixOps
+
+object IngressController extends Logger {
+
+  val ingressStrategy: IngressStrategy = {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala]:
``` 
 private lazy val ingressStrategy: IngressStrategy = {
    val client = new DefaultKubernetesClient()
    val versionInfo = client.getVersion
    val version = s"${versionInfo.getMajor}.${versionInfo.getMajor}".toDouble
    if (version >= 1.19) {
      new IngressStrategyV1()
    } else {
      new IngressStrategyV1beta1()
    }
  }
```

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressStrategyV1beta1.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          .withName("rest")
+          .endPort()
+          .endService()
+          .endBackend()
+          .endPath()
+          .endHttp()
+          .endRule()
+          .endSpec()
+          .build()
+        client.network.v1.ingresses().inNamespace(nameSpace).create(ingress)
```
Comment (senlizishi) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressStrategyV1beta1.scala]: There seems to be a reversal here. This is IngressStrategyV1beta1.scala, but it uses the API of Ingress v1

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      clusterId: String,
+      clusterClient: ClusterClient[_]): String = {
+    ingressStrategy.ingressUrlAddress(nameSpace, clusterId, clusterClient)
+  }
+
+  def prepareIngressTemplateFiles(buildWorkspace: String, ingressTemplates: String): String = {
+    ingressStrategy.prepareIngressTemplateFiles(buildWorkspace, ingressTemplates)
+  }
+
+  def getKubernetesVersion(): Double = {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala]: We can remove this method

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/BaseIngressStrategy.scala [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import org.json4s.jackson.JsonMethods.parse
+
+import java.io.File
+import java.nio.file.{Files, Paths}
+
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+abstract class BaseIngressStrategy extends IngressStrategy {
+  override def ingressUrlAddress(
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/BaseIngressStrategy.scala]: This method is not suitable for writing here, should be subclasses(IngressStrategyV1 | IngressStrategyV1beta1) to implement

**Comment by senlizishi** [comment]
LGTM

**Review by wolfboys — APPROVED** [review]
LGTM!

**Pull request merged**